python/nwm_client_new/src/hydrotools/nwm_client_new/NWMClient.py

```diff
@@ -60,20 +63,53 @@ class NWMClientDefaults:
         "streamflow": "value"
     })
     SSL_CONTEXT: ssl.SSLContext = ssl.create_default_context()
-    def __post_init__(self):
-        # Gather routelink files
-        rl_filepath = Path(__file__).parent / "data/routelink_files"
-        rl_files = rl_filepath.glob("*.csv")
-
-        # Generate crosswalk
-        dfs = []
-        for rl_file in rl_files:
-            dfs.append(pd.read_csv(
-            rl_file,
-            dtype={"nwm_feature_id": int, "usgs_site_code": str},
-            comment='#'
-        ).set_index('nwm_feature_id')[['usgs_site_code']])
-        self.CROSSWALK = pd.concat(dfs)
+    ROUTELINK_URL: str = "https://www.hydroshare.org/resource/d154f19f762c4ee9b74be55f504325d3/data/contents/RouteLink.h5"
+
+    def get_routelink(
+        self, 
+        url: str,
```

Comment:
I have a few comments/questions about this method:

1. since this method does not use anything from `self`, it might be better suited as a `staticmethod`.
2. I like the flexibility of the `url` parameter, but why not also have flexibility for the output file name? 
3. IMO, given that this method serves a single purpose (kudos!), it might make sense to give defaults for the keyword arguments (`url=NWMClientDefaults.ROUTELINK_URL`, and `output_name=NWMClientDefaults.ROUTELINK_OUTPUT_NAME` or something like that)

Reply:
I'm mulling this over. I'm not sure these three recommendations are compatible with each other.

Reply:
OK, it may not be what you had in mind, but I decided to err on the side of maintainability. I decided to rename `get_routelink` to `_download_and_read_routelink_file` and remove all parameters (except `self`). I think this is more suitable for an object that houses `NWMClientDefaults`. These values aren't really expected to be changed or used directly by a user (and should probably be spun off to their own module). In this vein, `NWMFileClient` should probably also be it's own module, but I'll leave these organizational concerns to a separate PR.

I initially thought about incorporating more methods to facilitate download and consumption of RouteLink files, but I think that's outside the scope of `NWMClientDefaults`. I'm also not sure what added functionality we could include. A preprocessed `pandas.DataFrame` is easily retrievable direct from the HDF5 file on HydroShare.

Reply:
>  I decided to rename `get_routelink` to `_download_and_read_routelink_file` and remove all parameters (except `self`). I think this is more suitable for an object that houses `NWMClientDefaults`. These values aren't really expected to be changed or used directly by a user (and should probably be spun off to their own module). In this vein, `NWMFileClient` should probably also be it's own module, but I'll leave these organizational concerns to a separate PR.

👍🏻🚀

> I initially thought about incorporating more methods to facilitate download and consumption of RouteLink files, but I think that's outside the scope of `NWMClientDefaults`.

I agree with your judgment. @christophertubbs might have a NOMADs crawler that we could incorporate in the future to do route link or parameter file "discovery", although I am not so keen on the idea. Just a thought though.